src/index/src/inverted_index/search/index_apply/predicates_apply.rs

```diff
@@ -0,0 +1,341 @@
+                    IndexNotFoundStrategy::ThrowError => {
+                        return IndexNotFoundSnafu { name }.fail();
+                    }
+                }
+            };
+
+            let fst = reader.fst(meta).await?;
+            let values = fst_applier.apply(&fst);
+
+            let mut mapper = FstValuesMapper::new(&mut *reader, meta);
+            let bm = mapper.map_values(&values).await?;
+
+            bitmap &= bm;
+        }
+
+        Ok(bitmap.iter_ones().collect())
+    }
+}
+
+impl PredicatesIndexApplier {
+    /// Constructs an instance of `PredicatesIndexApplier` based on a list of tag predicates.
+    /// Chooses an appropriate `FstApplier` for each tag based on the nature of its predicates.
+    ///
+    /// * `predicates`: A list of tag predicates.
+    /// * `options`: Options for reading an inverted index (e.g., how to handle missing index).
+    pub fn try_new(
+        mut predicates: Vec<(String, Vec<Predicate>)>,
+        options: ReadOptions,
+    ) -> Result<Self> {
+        let mut fst_appliers = Vec::with_capacity(predicates.len());
+
+        // InList predicates are applied first to benefit from higher selectivity.
+        let in_list_index = predicates
+            .iter_mut()
+            .partition_in_place(|(_, ps)| ps.iter().any(|p| matches!(p, Predicate::InList(_))));
+        let mut iter = predicates.into_iter();
+        for _ in 0..in_list_index {
+            let (tag_name, predicates) = iter.next().unwrap();
+            let fst_applier = Box::new(KeysFstApplier::try_from(predicates)?) as _;
+            fst_appliers.push((tag_name, fst_applier));
+        }
+
+        for (tag_name, predicates) in iter {
+            if predicates.is_empty() {
+                continue;
+            }
+            let fst_applier = Box::new(IntersectionFstApplier::try_from(predicates)?) as _;
+            fst_appliers.push((tag_name, fst_applier));
+        }
+
+        Ok(PredicatesIndexApplier {
+            fst_appliers,
+            options,
+        })
+    }
+
+    /// Creates a `BitVec` representing the full range of data in the index for initial scanning.
+    fn bitmap_full_range(metadata: &InvertedIndexMetas) -> BitVec {
+        let total_count = metadata.total_row_count;
+        let segment_count = metadata.segment_row_count;
+        let len = (total_count + segment_count - 1) / segment_count;
+        BitVec::repeat(true, len as _)
+    }
+}
+
+/// Options for reading an inverted index.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Default)]
+pub struct ReadOptions {
+    /// `index_not_found_strategy` controls the behavior of the applier when the index is not found.
+    pub index_not_found_strategy: IndexNotFoundStrategy,
+}
+
+/// Defines the behavior of an applier when the index is not found.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Default)]
+pub enum IndexNotFoundStrategy {
+    /// If the index is not found, the applier will return an empty list of indices.
```

Comment:
Could simplify the description of these variants. The struct already describes its purpose. For example:

```rust
/// Return an empty list of indices.
ReturnEmpty,

/// Return a list of all indices.
ReturnFullRange
...
}
```